hr_employee_service/models/hr_employee.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+from math import fabs
+from dateutil.relativedelta import relativedelta
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    service_hire_date = fields.Date(
+        string='Hire Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Hire date is normally the date an employee completes new hire'
+            ' paperwork'
+        ),
+    )
+    service_start_date = fields.Date(
+        string='Start Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Start date is the first day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_termination_date = fields.Date(
+        string='Termination Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Termination date is the last day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_duration = fields.Integer(
+        string='Service Duration',
+        readonly=True,
+        compute='_compute_service_duration',
+        help='Service duration in days',
+    )
+    service_duration_years = fields.Integer(
+        string='Service Duration (years)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+    service_duration_months = fields.Integer(
+        string='Service Duration (months)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+    service_duration_days = fields.Integer(
+        string='Service Duration (days)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+
+    @api.multi
+    @api.depends('service_start_date', 'service_termination_date')
+    def _compute_service_duration(self):
+        for record in self:
+            if record.service_start_date:
+                service_since = record.service_start_date
```

Comment:
Can we default to hire date if not service start date is given?

Reply:
Well, I'd avoid that since case when offer is signed yet start date is negotiated won't be handled properly. What are your thoughts on that?

Reply:
Start date and Hire date are the same thing, that's what I've seen in all HRMSs I worked with.
Contract signed date is something else.

Reply:
(there is still a point to give the possibility to manually give a "service start date" to use for service time. For example, if the worker was transferred from a related company/organization, and is entitled to keep his service time since the hire date in the origin organization).

Reply:
I'd still keep those two fields separate, while maybe prefilling in the UI the Start Date when Hire Date is altered and Start Date is empty. Can we call that a solution?

Reply:
It is, but I think it's easier to add an "or" to the computation start date, like you did for the termination date.

Reply:
Easier yet not transparent for the end-user (who does not read the code?) - let's try the via-UI approach